eclair-core/src/main/scala/fr/acinq/eclair/router/RouteCalculation.scala

```diff
@@ -69,11 +68,19 @@ object RouteCalculation {
         case PredefinedChannelRoute(targetNodeId, channels) =>
           val (end, hops) = channels.foldLeft((localNodeId, Seq.empty[ChannelHop])) {
             case ((start, current), shortChannelId) =>
-              d.channels.get(shortChannelId).flatMap(c => start match {
-                case c.ann.nodeId1 => g.getEdge(ChannelDesc(shortChannelId, c.ann.nodeId1, c.ann.nodeId2))
-                case c.ann.nodeId2 => g.getEdge(ChannelDesc(shortChannelId, c.ann.nodeId2, c.ann.nodeId1))
+              val channelDesc_opt = d.channels.get(shortChannelId).flatMap(c => start match {
+                case c.ann.nodeId1 => Some(ChannelDesc(shortChannelId, c.ann.nodeId1, c.ann.nodeId2))
+                case c.ann.nodeId2 => Some(ChannelDesc(shortChannelId, c.ann.nodeId2, c.ann.nodeId1))
                 case _ => None
```

Comment:
We tend to use pattern matching a lot in that kind of scenario, I personally find it slightly more readable than cascading ifs (but it's not a strong feeling).

When I first started working on eclair, I wasn't really used to it but just followed it for consistency since that was what the rest of the codebase tended to do, and now my brain got really used to reading those :nerd_face: